processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1736,9 +1736,12 @@ private ReportingPolicyGem getUnmappedSourcePolicy() {
             if ( mappingReferences.isForForgedMethods() ) {
                 return ReportingPolicyGem.IGNORE;
             }
-            // If we have ignoreByDefault = true, unprocessed source properties are not an issue.
-            if ( method.getOptions().getBeanMapping().isignoreByDefault() ) {
```

Comment:
I think that we can leave this as is and then just return 
```java
method.getOptions().getBeanMapping().unmappedSourcePolicy();
```

That makes sure that if it is not set on the bean mapping then the next one will be used.